### PR TITLE
Update EventGrid Doc Overview Page

### PR DESCRIPTION
**Pull request by YijunXieMS**
Update EventGrid ref doc to include the new version of SDK 4.x.x.

closes #21619

### CODE REVIEW

File: docs-ref-mapping/reference-latest.yml [hunk 1]
```diff
@@ -420,7 +420,6 @@
       landingPageType: Service
       children:
       - com.azure.messaging.eventgrid*
-      - com.microsoft.azure.eventgrid*
```
Comment (ramya-rao-a) [inline docs-ref-mapping/reference-latest.yml:423]: why are we removing this?

Comment (YijunXieMS) [reply docs-ref-mapping/reference-latest.yml:423]: This package name is for T1. But this is reference-latest.yml.

Comment (ramya-rao-a) [reply docs-ref-mapping/reference-latest.yml:423]: So, shouldnt this go into the reference-legacy file then?
Regardless, can we keep this for a few more months until we are done with the reliability improvements and mark T1 as deprecated?

Comment (YijunXieMS) [reply docs-ref-mapping/reference-latest.yml]: Added back.

**Comment by YijunXieMS** [comment]
> @YijunXieMS Please keep older version of eventgrid in the mappings. We'll merge the rest after you do so.

I added it back bro.

**Comment by scbedd** [comment]
Ok sir I'll merge this after the build completes.

**Pull request merged**